lib/driver.js

```diff
@@ -83,20 +74,54 @@ class AndroidDriver extends BaseDriver {
   }
 
   async initAndroidBootstrap () {
-    this.bootstrap = new Bootstrap(DEVICE_PORT);
+    this.bootstrap = new Bootstrap(DEVICE_PORT, this.opts.websocket);
 
-    await this.bootstrap.start(this.opts.appPackage, false);
+    await this.bootstrap.start(this.opts.appPackage, this.opts.disableAndroidWatchers);
   }
 
   async startAUT () {
     await this.adb.startApp({
       pkg: this.opts.appPackage,
       activity: this.opts.appActivity,
-      action: "android.intent.action.MAIN",
-      category: "android.intent.category.LAUNCHER",
-      flags: "0x10200000"
+      action: this.opts.intentAction,
+      category: this.opts.intentCategory,
+      flags: this.opts.intentFlags,
+      waitPkg: this.opts.appWaitPackage,
+      waitActivity: this.opts.appWaitActivity,
+      optionalIntentArguments: this.opts.optionalIntentArguments,
+      stopApp: this.opts.stopAppOnReset || !this.opts.dontStopAppOnReset
+      pkg: this.opts.appPackage,
+      activity: this.opts.appActivity,
     });
   }
+
+  async checkAppPresent () {
+    logger.debug("Checking whether app is actually present");
+    if (!(await fs.exists(this.opts.app))) {
+      logger.errorAndThrow(`Could not find app apk at ${this.opts.app}`);
+    }
+  }
+
+  async pushStrings (language) {
+    let remotePath = '/data/local/tmp';
+    let stringsJson = 'strings.json';
+    let stringsTmpDir = path.resolve(this.opts.tmpDir, this.opts.appPackage);
+    try {
+      let {apkStrings, localPath} = await this.adb.extractStringsFromApk(
+            this.opts.language, this.opts.app, stringsTmpDir);
+      this.apkStrings[this.opts.language] = apkStrings;
+      await this.adb.push(jsonFile, remotePath);
+    } catch (err) {
+      if (!(await fs.exists(this.opts.app))) {
+        // delete remote string.json if present
+        await this.adb.rimraf(`${remotePath}/${stringsJson}`);
+      } else {
+        log.warn("Could not get strings, continuing anyway");
+        let remoteFile = remotePath + '/' + stringsJson;
```

Comment:
can't you use string interpolation for this? then it will be the same as line 117.